ci broken: Build and Push 'iceberg-rest-fixture' Docker Image

### Apache Iceberg version

None

### Query engine

None

### Please describe the bug 🐞

A couple of failures from `main` branch run
https://github.com/apache/iceberg/actions/workflows/publish-iceberg-rest-fixture-docker.yml?query=branch%3Amain

Specifically https://github.com/apache/iceberg/actions/runs/20631392157/job/59249784049:
```
> Task :iceberg-open-api:processTestFixturesResources

/home/runner/work/iceberg/iceberg/api/src/main/java/org/apache/iceberg/io/RangeReadable.java:109: warning: [ByteBufferBackingArray] ByteBuffer.array() shouldn't be called unless ByteBuffer.arrayOffset() is used or if the ByteBuffer was initialized using ByteBuffer.wrap() or ByteBuffer.allocate().
> Task :iceberg-api:compileJava
      readFully(range.offset(), buffer.array());
                                            ^
    (see https://errorprone.info/bugpattern/ByteBufferBackingArray)
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
1 warning

> Task :iceberg-api:classes
> Task :iceberg-api:jar
> Task :iceberg-aws-bundle:jar SKIPPED
> Task :iceberg-gcp-bundle:jar SKIPPED
> Task :iceberg-core:compileJava FAILED

Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
> Task :iceberg-api:compileTestJava
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.


[Incubating] Problems report is available at: file:///home/runner/work/iceberg/iceberg/build/reports/problems/problems-report.html
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':iceberg-core:compileJava'.

> Could not resolve all files for configuration ':iceberg-core:compileClasspath'.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
   > Could not download zookeeper-3.8.4.jar (org.apache.zookeeper:zookeeper:3.8.4)

      > Could not get resource 'https://repo.maven.apache.org/maven2/org/apache/zookeeper/zookeeper/3.8.4/zookeeper-3.8.4.jar'.
         > Could not GET 'https://repo.maven.apache.org/maven2/org/apache/zookeeper/zookeeper/3.8.4/zookeeper-3.8.4.jar'. Received status code 403 from server: Forbidden
   > Could not download jakarta.activation-api-1.2.2.jar (jakarta.activation:jakarta.activation-api:1.2.2)
      > Could not get resource 'https://repo.maven.apache.org/maven2/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar'.
         > Could not HEAD 'https://repo.maven.apache.org/maven2/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar'. Received status code 403 from server: Forbidden
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.


* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org./

BUILD FAILED in 2m 49s
For more on this, please refer to https://docs.gradle.org/8.14.3/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 17 executed
Error: Process completed with exit code 1.
```

### Willingness to contribute

- [ ] I can contribute a fix for this bug independently
- [ ] I would be willing to contribute a fix for this bug with guidance from the Iceberg community
- [ ] I cannot contribute a fix for this bug at this time

--- a/.github/workflows/api-binary-compatibility.yml
+++ b/.github/workflows/api-binary-compatibility.yml
@@ -55,6 +55,13 @@ jobs:
         with:
           distribution: zulu
           java-version: 17
+      - uses: actions/cache@v5
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: ${{ runner.os }}-gradle-
       - run: |
           echo "Using the old version tag, as per git describe, of $(git describe)";
       - run: ./gradlew revapi --rerun-tasks

--- a/.github/workflows/publish-iceberg-rest-fixture-docker.yml
+++ b/.github/workflows/publish-iceberg-rest-fixture-docker.yml
@@ -42,6 +42,13 @@ jobs:
       with:
         distribution: zulu
         java-version: 21
+    - uses: actions/cache@v5
+      with:
+        path: |
+          ~/.gradle/caches
+          ~/.gradle/wrapper
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+        restore-keys: ${{ runner.os }}-gradle-
     - name: Build Iceberg Open API project
       run: ./gradlew :iceberg-open-api:shadowJar
     - name: Login to Docker Hub

--- a/.github/workflows/publish-snapshot.yml
+++ b/.github/workflows/publish-snapshot.yml
@@ -38,6 +38,13 @@ jobs:
         with:
           distribution: zulu
           java-version: 17
+      - uses: actions/cache@v5
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: ${{ runner.os }}-gradle-
       - run: |
           ./gradlew printVersion
           ./gradlew -DallModules publishApachePublicationToMavenRepository -PmavenUser=${{ secrets.NEXUS_USER }} -PmavenPassword=${{ secrets.NEXUS_PW }}

